content/blog/building-images-bake/index.md

```diff
@@ -0,0 +1,169 @@
+---
+title: "Creating a custom container image for CloudNativePG v2.0"
+date: 2025-06-17
+draft: false
+image:
+    url: 
+    attribution: 
+author: jgonzalez
+tags:
+ - blog
+ - information
+ - programming
+ - applications
+ - containers
+ - postgresql
+ - postgres
+ - images
+ - tutorial
+ - bake
+ - docker
+summary: Creating a container image for CloudNativePG Operator v2.0
+---
+
+## Summary
+In an almost [two years old blog post]({{% ref "/blog/creating-container-images/" %}}), we explained how
+to build a custom container image for CloudNativePG. After two years, many things have changed in the world of containers.
+One of those things has been the introduction of [Bake](https://docs.docker.com/build/bake/) in Docker, which allows you to build
+images using a simple configuration file. Bake is now our recommended way to build images for CloudNativePG.
+
+We will follow a simple baking recipe to create a custom container image or a set of container images, since Bake
+allows you to build multiple images at once in a simple way.
+
+## Ingredients
+
+- A Bake file. We will use the one provided in the [CloudNativePG repository](https://github.com/cloudnative-pg/postgres-containers/blob/main/docker-bake.hcl)
+- Another Bake file, but this time a local one, this one is to overwrite the previous one.
+
+Baking time: 5 minutes.
+
+## Instructions
+
+### Step 1: Prepare local Bake file
+
+In a local file with name [bake.hcl](bake.hcl), we add the following content, which is a simple Bake file that will build a custom image
+
+```hcl
+extensions = [
+  "pgvector",
+]
+target "myimage" {
+  dockerfile-inline = <<EOT
+ARG BASEIMAGE="ghcr.io/cloudnative-pg/postgresql:16.9-standard-bookworm"
+FROM $BASEIMAGE AS myimage
+ARG EXTENSIONS
+USER root
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends $EXTENSIONS && \
+    apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false && \
+    rm -rf /var/lib/apt/lists/* /var/cache/* /var/log/*
+USER 26
+EOT
+
+  matrix = {
+    tgt = [
+      "myimage"
+    ]
+    pgVersion = [
+      "16.9",
+      "17.5",
+    ]
+  }
+  name = "postgresql-${index(split(".",cleanVersion(pgVersion)),0)}-standard-bookworm"
+  target = "${tgt}"
+  args = {
+    BASE_IMAGE = "ghcr.io/cloudnative-pg/postgresql:${cleanVersion(pgVersion)}-standard-bookworm",
+    EXTENSIONS = "${getExtensionsString(pgVersion, extensions)}",
+  }
+}
+```
+
+There are a few things that we should remark here:
+
+- The `extensions` variable is a list of extensions that we want to include in the image. In our recipe we are using `pg_vector`,
+  but you can add any other extension that you want to include in the image.
+- The `dockerfile-inline` variable contains our Dockerfile definition, which cannot be used remotely. We will explain more about this later.
+- The `target` and the `tgt` have the same name, you can use whatever you want here as a name
```

Comment:
First of all, if you want to start the second clause with "you". Please add a period, not a comma.
Or, you can rephrase so the second clause is not as abrupt.